### DISCUSSION TITLE
analyzing ir remote code

### DISCUSSION THREAD

**Discussion by cybernard** [bengtmartensson/IrScrutinizer#501] [Q&A]
I have a broadcom rm4 pro, and captured codes via home assistant.  
Now I have a rather lengthy string of codes.
Is this something that your program can decode?
I am trying to decode the internal values so I can enter any temperature.

JgDKAIyTDzgPFA84DxQPFA8VDxQPNxAUDxQPFA84DzcQFA8UDxQQFA83EDcPFA84DxQPOA8UDzgPNw84DzcQNw84DzcPOA84DzcPOA83EDcPOA83DzgPNxA3DzgPNw8UEBQPFA8UEK2Okw8VDzcPFBA3DzgPNw84DxQPOA83EDcPFA8VDzcPOA83EDcPFA8VDzcPFBA3DxQQNw8UDxUPFA8UDxQQFA8UDxQQFA8UDxQPFQ8UDxQPFBAUDxQPFBAUDxQPOA83DzgPNxAADQU=

**Comment by bengtmartensson** [comment]
Sorry for not answering earlier.

The [current snapshot](https://github.com/bengtmartensson/IrScrutinizer/releases/tag/ci-build) can decode this format. (I do not think the current official 2.4.0  can do it.) Using the snapshot,  you can paste into the "Scrutinize signal" and you will then get 

````
Freq=38000Hz[+4919,-4919,+552,-1820,+552,-552,+552,-1820,+552,-552,+552,-552,+552,-552,+552,-552,+552,-1820,+552,-552,+552,-552,+552,-552,+552,-1820,+552,-1820,+552,-552,+552,-552,+552,-552,+552,-552,+552,-1820,+552,-1820,+552,-552,+552,-1820,+552,-552,+552,-1820,+552,-552,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-552,+552,-552,+552,-552,+552,-552,+552,-4919,+4919,-4919,+552,-552,+552,-1820,+552,-552,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-552,+552,-1820,+552,-1820,+552,-1820,+552,-552,+552,-552,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-552,+552,-552,+552,-1820,+552,-552,+552,-1820,+552,-552,+552,-1820,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-552,+552,-1820,+552,-1820,+552,-1820,+552,-1820,+552,-109456][][]
{38.0k,552,msb}<1,-1|1,-3>(9,-9,A:48,1,-9,9,-9,B:48,1,-109m){A=0xa1186afffff0,B=0x5ee79500000f}
````
which looks pretty plausible as AC signal. Does this help you?

**Reply by cybernard** [reply]
I am not sure,  there's a lot of -552 and positive 552 that have no meaning to me.
48 hex =72 so maybe that is the temperature.
I guess I will have to gather other codes and careful document what I changed before trying to decode it.

I would like to integrate this into home assistant so I can actually change my  AC settings without storing 100's of the above codes.

**Comment by bengtmartensson** [comment]
First step is to analyze the IR protocol (or at least as much as you need). As you see from my previous post, IrScrutinizer "decrypts" the code to `{38.0k,552,msb}<1,-1|1,-3>(9,-9,A:48,1,-9,9,-9,B:48,1,-109m){A=0xa1186afffff0,B=0x5ee79500000f}`, so "all" you have to do is to analyze sufficiently many sufficiently chosen IR signal to find out e.g. exactly where the temperature etc. sits. Next step is to have HA generate and send these signals on request. This is somewhat hairy...  

> I am not sure, there's a lot of -552 and positive 552 that have no meaning to me.
 
Negative numbers are gaps, here of 552 micro seconds, and positive ones are pulses, i.e., when the carrier is on.
